ExampleCode/C++/API/API.cpp

```diff
@@ -0,0 +1,916 @@
+      // if the JSON array ended up empty. Probably wasn't any projects with that search term.
+      cout << "\nProject title array is empty.\n";
+
+      // Clean up cURL and close the memfile
+      curl_easy_cleanup(curl);
+      curl_global_cleanup();
+      memfclose(json_file);
+
+      return project_titles;  // this is an empty vector
+    }
+
+    // If we make here, we can make a vector of project titles and return that to the user.
+    for(it = projects_array.begin(); it != projects_array.end(); it++)
+    {
+      // Get the current object
+      object obj = it->get<object>();
+
+      // Grab the field name
+      string name = obj["name"].get<string>();
+
+      // Push the name back into the vector.
+      project_titles.push_back(name);
+    }
+
+    // Clean up cURL and close the memfile
+    curl_easy_cleanup(curl);
+    curl_global_cleanup();
+    memfclose(json_file);
+
+    return project_titles;
+  }
+}
+
+
+// Checks to see if the email / password is valid
+bool iSENSE::get_check_user()
+{
+  // First check to see if the email and password have been set.
+  if(email == "email" || password == "password")
+  {
+    cout << "Please set an email & password for this project.\n";
+  }
+  else if(email.empty() == true || password.empty() == true)
+  {
+    cout << "Please set an email & password for this project.\n";
+  }
+
+  // If we get here, an email and password have been set, so do a GET using
+  // the email & password.
+
+  get_UserURL = devURL + "/users/myInfo?email=" + email + "&password=" + password;
+
+  // This project will try using CURL to make a basic GET request to rSENSE
+  CURL *curl = curl_easy_init();          // cURL object
+  CURLcode curl_code;                     // cURL status code
+  long http_code;                         // HTTP status code
+
+  if(curl)
+  {
+    // Set the GET URL, in this case the one be created above using the user's
+    // email & password.
+    curl_easy_setopt(curl, CURLOPT_URL, get_UserURL.c_str());
+
+    // Disable libcURL's desire to output to the screen by giving it a
+    // write function which basically does nothing but returns the size of what
+    // would have been outputted.
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, suppress_output);
+
+    // Perform the request
+    curl_easy_perform(curl);
+
+    // We can actually get the HTTP response code from cURL, so let's do so to check for errors.
+    http_code = 0;
+
+    // This will put the HTTP response code into the "http_code" variable.
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+
+    /*
+     *                The iSENSE API gives us two response codes to check against:
+     *                Success: 200 OK
+     *                Failure: 401 Unauthorized
+     */
+
+    if(http_code == 200)
+    {
+      // Clean up cURL
+      curl_easy_cleanup(curl);
+      curl_global_cleanup();
+
+      // Return success.
+      return true;
+    }

```

Comment:
Lines 302 - 321:
Since you're returning `true` in this if statement, you don't use an `else` block that runs lines 311-321.  So something like the following would suffice:

```
+    if(http_code == 200)
+    {
+      // Clean up cURL
+      curl_easy_cleanup(curl);
+      curl_global_cleanup();
+
+      // Return success.
+      return true;
+    }
+    cout << "\nThe email and/or password you entered was **not** valid.\n";
+    cout << "Please try entering the email / password again.\n";
+
+    // Clean up cURL
+    curl_easy_cleanup(curl);
+    curl_global_cleanup();
+
+    // Return failure.
+    return false;
```